File: ios/Classes/plugin/di/AppModule.swift
import Foundation

// Modulo de injeção de dependência para o aplicativo
public class AppModule {
    // Instâncias do BluetoothDataSource e BluetoothRepository
    private static let bluetoothDataSource = BluetoothDataSource()
    private static let bluetoothRepository: BluetoothRepository = BluetoothRepositoryImpl(dataSource: bluetoothDataSource)
    
    // Casos de uso
    public static let scanDevicesUseCase = ScanDevicesUseCase(repository: bluetoothRepository)
    public static let connectDeviceUseCase = ConnectDeviceUseCase(repository: bluetoothRepository)
    public static let printUseCase = PrintUseCase(repository: bluetoothRepository)
    public static let disconnectDeviceUseCase = DisconnectDeviceUseCase(repository: bluetoothRepository)
    public static let feedLineUseCase = FeedLineUseCase(repository: bluetoothRepository)
}

File: android/src/main/kotlin/com/maktubcompany/easy_blue_printer/plugin/di/AppModule.kt
package com.maktubcompany.easy_blue_printer.plugin.di

import com.maktubcompany.easy_blue_printer.plugin.data.datasource.BluetoothDataSource
import com.maktubcompany.easy_blue_printer.plugin.data.repository.BluetoothRepositoryImpl
import com.maktubcompany.easy_blue_printer.plugin.domain.repository.BluetoothRepository
import com.maktubcompany.easy_blue_printer.plugin.domain.usecase.ConnectDeviceUseCase
import com.maktubcompany.easy_blue_printer.plugin.domain.usecase.DeviceIsConneceted
import com.maktubcompany.easy_blue_printer.plugin.domain.usecase.DisconnectDeviceUseCase
import com.maktubcompany.easy_blue_printer.plugin.domain.usecase.FeedLineUseCase
import com.maktubcompany.easy_blue_printer.plugin.domain.usecase.PrintUseCase
import com.maktubcompany.easy_blue_printer.plugin.domain.usecase.GetPairedDevices
import com.maktubcompany.easy_blue_printer.plugin.domain.usecase.PrintImageUseCase

object AppModule {
    private val bluetoothDataSource = BluetoothDataSource()
    private val bluetoothRepository: BluetoothRepository = BluetoothRepositoryImpl(bluetoothDataSource)

    val getPairedDevices = GetPairedDevices(bluetoothRepository)
    val connectDeviceUseCase = ConnectDeviceUseCase(bluetoothRepository)
    val printUseCase = PrintUseCase(bluetoothRepository)
    val disconnectDeviceUseCase = DisconnectDeviceUseCase(bluetoothRepository)
    val feedLineUseCase = FeedLineUseCase(bluetoothRepository)
    val isConnectedUseCase = DeviceIsConneceted(bluetoothRepository)
    val printImageUseCase = PrintImageUseCase(bluetoothRepository)
}